src/coord/src/coord.rs

```diff
@@ -1129,7 +1143,10 @@ impl<S: Append + 'static> Coordinator<S> {
                     .expect("sending to internal_cmd_tx cannot fail");
             });
         } else {
-            self.group_commit().await;
+            match Arc::clone(&self.write_lock).try_lock_owned() {
```

Comment:
I tried to get `try_lock` to work, but ran into a borrow checker issue. With `try_lock` you end up with a guard that borrows the mutex which requires an immutable self reference. Then `self.group_commit` requires a mutable self reference.
Here's the compiler error:
```
error[E0502]: cannot borrow `*self` as mutable because it is also borrowed as immutable
    --> src/coord/src/coord.rs:1148:31
     |
1146 |             let res = self.write_lock.try_lock();
     |                       -------------------------- immutable borrow occurs here
1147 |             match res {
1148 |                 Ok(_guard) => self.group_commit().await,
     |                               ^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
...
1151 |         }
     |         - immutable borrow might be used here, when `res` is dropped and runs the destructor for type `std::result::Result<tokio::sync::MutexGuard<'_, ()>, tokio::sync::TryLockError>`

error[E0502]: cannot borrow `*self` as mutable because it is also borrowed as immutable
    --> src/coord/src/coord.rs:1149:27
     |
1146 |             let res = self.write_lock.try_lock();
     |                       -------------------------- immutable borrow occurs here
...
1149 |                 Err(_) => self.defer_write(Deferred::GroupCommit),
     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
1150 |             };
1151 |         }
     |         - immutable borrow might be used here, when `res` is dropped and runs the destructor for type `std::result::Result<tokio::sync::MutexGuard<'_, ()>, tokio::sync::TryLockError>`

```